api/src/search/backend/load_opportunities_to_index.py

```diff
@@ -68,21 +72,61 @@ def run_task(self) -> None:
     def incremental_updates_and_deletes(self) -> None:
         existing_opportunity_ids = self.fetch_existing_opportunity_ids_in_index()
 
-        # load the records incrementally
-        # TODO - The point of this incremental load is to support upcoming work
-        #        to load only opportunities that have changes as we'll eventually be indexing
-        #        files which will take longer. However - the structure of the data isn't yet
-        #        known so I want to hold on actually setting up any change-detection logic
-        loaded_opportunity_ids = set()
-        for opp_batch in self.fetch_opportunities():
-            loaded_opportunity_ids.update(self.load_records(opp_batch))
+        # Fetch opportunities that need processing from the queue
+        queued_opportunities = (
+            self.db_session.execute(
+                select(Opportunity)
+                .join(OpportunitySearchIndexQueue)
+                .join(CurrentOpportunitySummary)
+                .where(
+                    Opportunity.is_draft.is_(False),
+                    CurrentOpportunitySummary.opportunity_status.isnot(None),
+                )
+                .options(selectinload("*"), noload(Opportunity.all_opportunity_summaries))
+            )
+            .scalars()
+            .all()
+        )
 
-        # Delete
-        opportunity_ids_to_delete = existing_opportunity_ids - loaded_opportunity_ids
+        # Process updates and inserts
+        processed_opportunity_ids = set()
+        opportunities_to_index = []
 
-        if len(opportunity_ids_to_delete) > 0:
+        for opportunity in queued_opportunities:
+            logger.info(
+                "Processing queued opportunity",
+                extra={
+                    "opportunity_id": opportunity.opportunity_id,
+                    "in_search": opportunity.opportunity_id in existing_opportunity_ids,
```

Comment:
What about something like this to get update/insert
```suggestion
                    "status": "update" if opportunity.opportunity_id in existing_opportunity_ids else "insert",
```